### PR TITLE
Implemented unit tests

### PR DESCRIPTION
**Pull request by vangelif**
In this PR, @luigirazum @vangelif, we:

- Added unit tests for all our classes
- Used Rspec composing our unit tests
- Fixed all linter errors
- Used correct Gitflow
- Confirmed that documentation is concise and professional

### CODE REVIEW

File: spec/teacher_spec.rb [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+  it 'creates a teacher object with name, age, and specialization' do
+    teacher = Teacher.new(name: name, age: age, specialization: specialization)
+    expect(teacher.name).to eq(name)
+    expect(teacher.age).to eq(age)
+    expect(teacher.specialization).to eq(specialization)
+  end
+  it 'sets id to nil if not provided' do
+    teacher = Teacher.new(name: name, age: age, specialization: specialization)
+    expect(teacher.id).to be_nil
+  end
```
Comment (codepantha) [inline spec/teacher_spec.rb]: - [x] This test is failing as you can see below 👇 Please try to make it green 🟢 
image

File: spec/book_spec.rb [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+  end
+
+  it 'title book equals to title one' do
+    book = Book.new(title: 'title one', author: 'an author')
+    expect(book.title).to eql 'title one'
+  end
+
+  it 'author equals to an author' do
+    book = Book.new(title: 'title one', author: 'an author')
+    expect(book.author).to eql 'an author'
```
Comment (codepantha) [inline spec/book_spec.rb:18]: - [x] According to project requirements, we should write tests for all the methods in all the classes. In this case, please write tests for:
1. `add_rental`
2. `to_json`
3. `to_s`

File: spec/classroom_spec.rb [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+describe Classroom do
+  let(:classroom_label) { 'Ruby & RSpec' }
+  let(:classroom) { Classroom.new(classroom_label) }
+
+  it 'creates a new Classroom object with the given label' do
+    expect(classroom.label).to eq(classroom_label)
+  end
+
+  it 'initializes students as an empty array' do
```
Comment (codepantha) [inline spec/classroom_spec.rb]: - [ ] Please write unit tests for the `add_student` method in the `Classroom` class 👍

File: spec/person_spec.rb [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+  # Test 1: Check if the person's age is set correctly
+  it 'has the correct age' do
+    expect(@person.age).to eq(25)
+  end
+
+  # Test 2: Check if the person can use services when of age
+  it 'can use services when of age' do
+    expect(@person.can_use_services?).to be(true)
+  end
+
```
Comment (codepantha) [inline spec/person_spec.rb:17]: - [x] Please write unit tests for the `add_student` method in the `Person` class 👍

File: spec/rental_spec.rb [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+describe Rental do
+  describe 'creates a Rental instance' do
+    it 'the result is an instance of Rental class' do
+      person = Teacher.new(name: 'Mr. Smith', age: 35, specialization: 'Chemistry')
+      book = Book.new(title: 'Chemistry for newbies', author: 'The teacher')
+      rental = Rental.new('2023-08-02', book, person)
+      expect(rental).to be_an_instance_of Rental
+    end
+
+    it 'the date attribute for the rental is "2023-08-02"' do
```
Comment (codepantha) [inline spec/rental_spec.rb:14]: - [ ] According to project requirements, we should write tests for all the methods in all the classes. In this case, please write tests for:
1. `to_json`
2. `to_s`

File: spec/student_spec.rb [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+    it 'the name attribute for the student is "John"' do
+      student = Student.new(name: 'John', age: 14)
+      expect(student.name).to eql 'John'
+    end
+
+    it 'the age attribute for the student is 14' do
+      student = Student.new(name: 'John', age: 14)
+      expect(student.age).to eql 14
+    end
+
```
Comment (codepantha) [inline spec/student_spec.rb]: - [ ] According to project requirements, we should write tests for all the methods in all the classes. In this case, please write tests for:
1. `add_classroom`
2. `to_json`
3. `to_s`
4. `play_hooky`

**Review by codepantha — CHANGES_REQUESTED** [review]
# STATUS; CHANGES REQUIRED ♻️ 

## Hi team 👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights 🎯 
✔️ Descriptive PR
✔️ Good commit history
✔️ Wrote unit tests for most methods in your classes 📌 
✔️ No linter errors 🟢 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- n/a

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/teacher_spec.rb (hunk 1 above)
Comment (luigirazum) [reply spec/teacher_spec.rb]: Test for 'Teacher' class were fixed in [this commit](https://github.com/vangelif/ruby_library/pull/7/commits/a40f3d3a825250f6fc54f2748ba5696b894d8a70)

image

File: spec/person_spec.rb (hunk 4 above)
Comment (vangelif) [reply spec/person_spec.rb:17]: According to project requirements, we should write unit tests for all methods in Person class. 
Above method doesn't exist in subject class.

File: spec/rental_spec.rb (hunk 5 above)
Comment (luigirazum) [reply spec/rental_spec.rb:14]: Tests for `to_s` and `to_json` methods were added in [this commit](https://github.com/vangelif/ruby_library/pull/7/commits/9641fed8d78f223450a947c0676f3003245fb6c7)

image

File: spec/student_spec.rb (hunk 6 above)
Comment (luigirazum) [reply spec/student_spec.rb]: - `add_classroom` tests were added with [this commit](https://github.com/vangelif/ruby_library/pull/7/commits/a7e42234ab7a6c57e74642622a79e90b812bb5aa)
- `to_json` tests were added with [this commit](https://github.com/vangelif/ruby_library/pull/7/commits/64e56805d7d879ff98088fbc99a7c71965fe737d)
- `to_s` tests were added with [this commit](https://github.com/vangelif/ruby_library/pull/7/commits/a7e42234ab7a6c57e74642622a79e90b812bb5aa)
- `play_hooky` tests were added with [this commit](https://github.com/vangelif/ruby_library/pull/7/commits/4e876b0011b9d6fcea4217cbc56c4b70d87f321d)

image

**Review by ghost — APPROVED** [review]
Hi Team, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights

✅ Correct use of Gitflow
✅ No linter error
✅ App is well tested 👍 

## Optional suggestions

N/A

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**